lib/persistence/mongo/Base.js

```diff
@@ -0,0 +1,115 @@
+var Q = require("q");
+var mongo_promise = require('./client.js');
+var crypto = require('crypto');
+var RecordNotFoundError = require('./RecordNotFound')
+
+module.exports = function(options){
+
+  var name = options['name'];
+  if ( !name ) { throw new Error("name is required"); }
+
+  var model = options['model'];
+  if ( !model ) { throw new Error("model is required"); }
+
+  var mongo_model = require('../backbone_sync.js')(model)
+
+  mongo_model = mongo_model.extend({
+
+    pk: function(){
+      return this.get('_id');
+    },
+
+    idAttribute: "_id",
+
+    create: function(attributes, options){
+
+      if(!this.collection_name){
+        throw new Error("need the collection_name attribute on the model!")
+      }
+
+      // 24 character id string, as opposed to using mongo's builtin ObjectID
+      this.set('_id', crypto.randomBytes(12).toString('hex'))
```

Comment:
I think the main reason is that the mongo [ObjectId](http://docs.mongodb.org/manual/reference/object-id/) is not a simple string in the Node Mongo library. You have to explicitly serialize and deserialize the ObjectId instance to a string for Backbone to consume.  It was a little bit cumbersome doing this for the crud operations

The other aspect of this, is that for some operations like `create` and `read` we are returning the javascript object result of the lookup, not the backbone model instance. So those cases we would have to intercept the ObjectID's anyways and turn them into strings. 

It seems easier to consider the id to be a unique string and that's it.